AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSCognitoUserPoolTokens.swift

```diff
@@ -38,6 +39,38 @@ public struct AWSCognitoUserPoolTokens: AuthCognitoTokens {
         self.expiration = expiration
```

Comment:
sorry i may be missing some context here, if `expiration` is deprecated, then i think the init `init(idToken:accessToken:refreshToken:expiration)` should also be deprecated. Then the new init that takes in `expiresIn` should hold/calculate the value from the claims "exp" in a new property called `expiresIn`? Introducing a new property will allow us to remove the deprecated one in a major version bump. Or perhaps `expiration` shouldn't be deprecated but only the `init(expiration)` should be, ie.
```
@available(*, deprecated, message: "Use of `init(idToken,accessToken,refreshToken:expiration)` is deprecated, use `exp` claim in the `idToken` or `accessToken` instead")
public init(idToken: String,
                accessToken: String,
                refreshToken: String,
                expiresIn: Int)
```

Reply:
let me know if this is off, my current understanding is
- I think `expiration` property shouldn't be deprecated because we are still setting the value, calculated from the claims. 
- the `expiration` property can still be read by the customer if they wanted to, but the init available to customer should be `init(idToken:accessToken:refreshToken)` without `expiresIn` property. this init does the caluclation and sets `self.expiration = ` directly

Reply:
I am gonna mark the methods as deprecated but not in favor of adding a new property because we are not relying on it any longer. Will be removed in the next major version.. This is a workaround for the customer to unblock them.